src/ml/neural_net/mps_descriptor_utils.h

```diff
@@ -0,0 +1,88 @@
+#import <Accelerate/Accelerate.h>
+#import <Foundation/Foundation.h>
+#import <Metal/Metal.h>
+#import <MetalPerformanceShaders/MetalPerformanceShaders.h>
+
+@interface TCMPSConvolutionDescriptor:NSObject
+@property (nonatomic) NSUInteger kernelWidth;
+@property (nonatomic) NSUInteger kernelHeight;
+@property (nonatomic) NSUInteger inputFeatureChannels;
+@property (nonatomic) NSUInteger outputFeatureChannels;
+@property (nonatomic) NSUInteger strideWidth;
+@property (nonatomic) NSUInteger strideHeight;
+@property (nonatomic) NSUInteger paddingWidth;
+@property (nonatomic) NSUInteger paddingHeight;
+@property (nonatomic, copy) NSString *label;
+@property (nonatomic) BOOL updateWeights;
+@end
+
+@interface TCMPSInstanceNormalizationDescriptor:NSObject
+@property (nonatomic) NSUInteger channels;
+@property (nonatomic) NSUInteger styles;
+@property (nonatomic, copy) NSString *label;
+@end
+
+@interface TCMPSUpsamplingDescriptor:NSObject
+@property (nonatomic) NSUInteger scale;
+@end
+
+@interface TCMPSPoolingDescriptor:NSObject
+@property (nonatomic) NSUInteger kernelSize;
+@property (nonatomic) NSUInteger strideSize;
+@end
+
+@interface TCMPSEncodingDescriptor:NSObject
+@property (nonatomic) TCMPSConvolutionDescriptor* conv;
+@property (nonatomic) TCMPSInstanceNormalizationDescriptor* inst;
+@end
+
+@interface TCMPSResidualDescriptor:NSObject
+@property (nonatomic) TCMPSConvolutionDescriptor* conv_1;
+@property (nonatomic) TCMPSConvolutionDescriptor* conv_2;
+@property (nonatomic) TCMPSInstanceNormalizationDescriptor* inst_1;
+@property (nonatomic) TCMPSInstanceNormalizationDescriptor* inst_2;
+@end
+
+@interface TCMPSDecodingDescriptor:NSObject
+@property (nonatomic) TCMPSConvolutionDescriptor* conv;
+@property (nonatomic) TCMPSInstanceNormalizationDescriptor* inst;
+@property (nonatomic) TCMPSUpsamplingDescriptor* upsample;
+@end
+
+@interface TCMPSBlock1Descriptor:NSObject
```

Comment:
A lot of these types seem ST-specific but have generic names. What should someone understand "Block1" to mean, especially outside of an ST context?

Reply:
So it's not specific to the Style Transfer Network, rather the VGG16 network. I can pre-fix them by that.